app/models/ability.rb

```diff
@@ -51,6 +51,9 @@ def initialize(user)
       can [:print_applications], Event
       can :manage, Request
       cannot [:check], ApplicationLetter
+      cannot :show, Profile, user: { id: !user.id }
+      cannot :view_personal_details, ApplicationLetter, user: { id: !user.id }
+      cannot :check, ApplicationLetter
```

Comment:
Could this restrictions be moved to pupil? I think they would then also be valid for coach?